File: problems/basic/53/solution2.cpp
// Accepted: 82ms (67.23%), 70.25MB (9.95%)

class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int curSum = nums[0];
        int maxSum = curSum;

        for (int i = 1; i < nums.size(); ++i) {
            curSum = max(curSum+nums[i], nums[i]);
            maxSum = max(maxSum, curSum);
        }

        return maxSum;
    }
};

File: problems/basic/53/solution2.py
# Accepted: 570ms (50.91%), 30.67MB (71.51%)

class Solution:
    def maxSubArray(self, nums):
        curSum = nums[0]
        maxSum = curSum

        for i in range(1, len(nums)):
            curSum = max(nums[i], curSum + nums[i])
            maxSum = max(maxSum, curSum)

        return maxSum

